console/network/src/lib.rs

```diff
@@ -118,7 +118,9 @@ pub trait Network:
     /// The cost in microcredits per constraint for the deployment transaction.
     const SYNTHESIS_FEE_MULTIPLIER: u64 = 25; // 25 microcredits per constraint
     /// The maximum number of constraints in a deployment.
-    const MAX_DEPLOYMENT_LIMIT: u64 = 1 << 20; // 1,048,576 constraints
+    const MAX_DEPLOYMENT_CONSTRAINTS: u64 = 1 << 20; // 1,048,576 constraints
```

Comment:
if the limit is `1 << 20`, is it reasonable to assume that we would never expect `1 << 64` to be supported? `u32` might be a better pick in that case, and IIRC that could have a performance impact on some of the R1CS operations (if the `Variable` was also capped at `1 << 32`

Reply:
Constraints won't exceed u32 anytime soon, especially as our SRS only supports `1 << 28`. Perhaps constant variables might end up being more. The `circuit` crate stores most data as u32, and its only a small amount of counters anyway, so at this point its not worth the rewrite I think.